spec/lib/invoice_generator_spec.rb

```diff
@@ -77,11 +92,12 @@ def check_invoice_for_single_vm(invoices, project, vm, duration, begin_time)
       ["project_name", project.name],
       ["billing_info", expected_billing_info],
       ["issuer_info", expected_issuer],
+      ["vat_info", expected_vat_info],
       ["resources", expected_resources],
       ["subtotal", expected_cost],
       ["discount", 0],
       ["credit", 0],
-      ["cost", expected_cost]
+      ["cost", (expected_cost + (expected_vat_info&.[]("amount") || 0)).round(3)]
```

Comment:
Up to you, but you may want to consider `expected_vat_info&.fetch("amount", 0)`, since the fallback value is an immediate.

Reply:
If `expected_vat_info` is nil, `expected_vat_info&.fetch("amount", 0)` returns nil and fails with `nil can't be coerced into Float`. I added `.to_f`